manifests/charts/istio-control/istio-discovery/templates/zzz_profile.yaml

```diff
@@ -41,3 +41,6 @@ Finally, we can set all of that under .Values so the chart behaves without aware
 {{- $a := mustMergeOverwrite $defaults $globals  }}
 {{- end }}
 {{- $b := set $ "Values" (mustMergeOverwrite $defaults $.Values) }}
+{{- with $.Values._internal_do_not_set.flatten }}
```

Comment:
Yes, this is ~roughly the same concept as 

https://github.com/istio/istio/pull/51517/files#diff-9e517c2405d381930226cbdc5b4ec16976f5c4939bc34bc145214a71df658be2R19

I really don't like using a user-facing value as a key to trigger the munge tho, that's spreading the fix around a bit too much IMO - I would prefer to just hardcode `pilot`/`cni` as descope targets here (which should work just as well)

Otherwise this general pattern (flattening the legacy chart nestings in `zzz_profile.yaml`) seems fine to me and more or less in line with what I gestured at in #51517

Reply:
if not doing this I would put it in the makefile like ztunnel personally. Probably both this + the ztunnel thing should do the same of the three options (internal values.yaml, makefile , or put chert-specific semantics into Zzz_profile), though

Reply:
Not sure if you were suggesting this not not, but just always flattening pilot and cni (without doing it only for those charts) *might* cause issues since `pilot.cni` is a valid setting today. That means, for the Istiod chart, `cni` is now a valid value 

If we descoped CNI in the pilot chart I suspect that would break

Reply:
> Not sure if you were suggesting this not not, but just always flattening pilot and cni (without doing it only for those charts) might cause issues since pilot.cni is a valid setting today. That means, for the Istiod chart, cni is now a valid value. 
> 
> If we descoped CNI in the pilot chart I suspect that would break

Yes possibly - this kind of thing is why I favor doing this for a few values at a time (labels/annos/images) and seeing what ends up being left.

If `pilot.cni` is a problem uniquely tho, we can just special case it when we de-scope `pilot` (since `cni.pilot` is, I very much hope, not also valid)

Reply:
I do not, in general, mind "ugly" descoping logic.

I would prefer it to live in one spot as much as possible, however, so that it is easier to understand and (eventually, maybe) remove.